src/java/org/apache/cassandra/db/virtual/SettingsTable.java

```diff
@@ -19,103 +19,149 @@
+        if (value == null)
+            throw new InvalidRequestException(String.format("Setting the value to null is equivalent to the deletion operation. " +
+                                              "Column deletion is not supported by the '%s.%s'", metadata().keyspace, metadata().name));
+        else
+            runExceptionally(() -> setPropertyFromString(getKeyAndWarnIfObsolete(name), value),
+                             t -> invalidRequest("Invalid update request '%s'. Cause: %s", name, t.getMessage()));
     }
 
     @Override
     public DataSet data(DecoratedKey partitionKey)
     {
         SimpleDataSet result = new SimpleDataSet(metadata());
         String name = UTF8Type.instance.compose(partitionKey.getKey());
-        if (BACKWARDS_COMPATABLE_NAMES.containsKey(name))
-            ClientWarn.instance.warn("key '" + name + "' is deprecated; should switch to '" + BACKWARDS_COMPATABLE_NAMES.get(name) + "'");
-        if (PROPERTIES.containsKey(name))
-            result.row(name).column(VALUE, getValue(PROPERTIES.get(name)));
+        try
+        {
+            Object value = getPropertyAsString(getKeyAndWarnIfObsolete(name));
+            result.row(name).column(VALUE, value).column(MUTABLE, DatabaseDescriptor.isMutableProperty(name));
+        }
+        catch (PropertyNotFoundException e)
+        {
+            // ignore
+        }
         return result;
     }
 
     @Override
     public DataSet data()
     {
         SimpleDataSet result = new SimpleDataSet(metadata());
-        for (Map.Entry<String, Property> e : PROPERTIES.entrySet())
-            result.row(e.getKey()).column(VALUE, getValue(e.getValue()));
+        for (String key : DatabaseDescriptor.getAllProperties())
+        {
+            if (BACKWARDS_COMPATABLE_NAMES.containsKey(key))
+                continue;
+            runExceptionally(() -> result.row(key)
+                                         .column(VALUE, getPropertyAsString(key))
+                                         .column(MUTABLE, DatabaseDescriptor.isMutableProperty(key)),
+                             t -> new ConfigurationException(t.getMessage(), false));
+        }
+
+        runExceptionally(() -> BACKWARDS_COMPATABLE_NAMES.forEach((oldName, newName) ->
+                                           result.row(oldName)
+                                                 .column(VALUE, getPropertyAsString(newName))
+                                                 .column(MUTABLE, DatabaseDescriptor.isMutableProperty(newName))),
+                         t -> new ConfigurationException(t.getMessage(), false));
         return result;
     }
 
-    private String getValue(Property prop)
+    static String getKeyAndWarnIfObsolete(String name)
     {
-        Object value = prop.get(config);
-        return value == null ? null : value.toString();
+        String key = BACKWARDS_COMPATABLE_NAMES.getOrDefault(name, name);
+        if (BACKWARDS_COMPATABLE_NAMES.containsKey(name))
```

Comment:
@Mmuzaf can't be this rewritten like `if (name.equals(key))`? So we dont need to look into map.